src/Aspire.Hosting.SurrealDb/SurrealDbBuilderExtensions.cs

```diff
@@ -0,0 +1,84 @@
+    /// <param name="userName">The parameter used to provide the administrator username for the SurrealDB resource.</param>
+    /// <param name="password">The parameter used to provide the administrator password for the SurrealDB resource. If <see langword="null"/> a random password will be generated.</param>
+    /// <param name="port">The host port for the SurrealDB instance.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<SurrealDbServerResource> AddSurrealServer(
+        this IDistributedApplicationBuilder builder,
+        string name,
+        IResourceBuilder<ParameterResource>? userName = null,
+        IResourceBuilder<ParameterResource>? password = null,
+        int? port = null
+    )
+    {
+        // The password must be at least 8 characters long and contain characters from three of the following four sets: Uppercase letters, Lowercase letters, Base 10 digits, and Symbols
+        var passwordParameter = password?.Resource ?? ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-password", minLower: 1, minUpper: 1, minNumeric: 1);
+
+        var surrealServer = new SurrealDbServerResource(name, userName?.Resource, passwordParameter);
+        return builder.AddResource(surrealServer)
+                      .WithEndpoint(port: port, targetPort: 8000, name: SurrealDbServerResource.PrimaryEndpointName)
+                      .WithImage(SurrealDbContainerImageTags.Image, SurrealDbContainerImageTags.Tag)
+                      .WithEnvironment(context =>
+                      {
+                          context.EnvironmentVariables[UserEnvVarName] = surrealServer.UserNameReference;
+                          context.EnvironmentVariables[PasswordEnvVarName] = surrealServer.PasswordParameter;
+                      })
+                      .WithEntrypoint("/surreal")
+                      .WithArgs("start", "--auth");
+    }
+
+    /// <summary>
+    /// Adds a SurrealDB database to the application model. This is a child resource of a <see cref="SurrealDbServerResource"/>.
+    /// </summary>
+    /// <param name="builder">The SurrealDB resource builders.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency.</param>
+    /// <param name="namespaceName">The name of the namespace. If not provided, this defaults to "test".</param>
+    /// <param name="databaseName">The name of the database. If not provided, this defaults to "test".</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<SurrealDbDatabaseResource> AddDatabase(this IResourceBuilder<SurrealDbServerResource> builder, string name, string namespaceName = "test", string databaseName = "test")
+    {
+        builder.Resource.AddDatabase(name, databaseName);
```

Comment:
Whats for?
```suggestion
        
```

Reply:
Taken from the SqlServer hosting package. It seems that this offers a way to query the list of database names inside this resource. Better change it to also hold the namespace name.

Reply:
Updated to have both `Namespaces` prop for server resource and `Databases` prop for namespace resource.